exporters/prometheus/src/exporter_utils.cc

```diff
@@ -324,6 +401,51 @@ void PrometheusExporterUtils::SetMetricBasic(prometheus_client::ClientMetric &me
           << "'. Ignoring this label.");
     }
   }
+
+  // Convert resource to job and instance labels
+  if (nullptr != resource)
+  {
+    do
+    {
+      opentelemetry::sdk::resource::ResourceAttributes::const_iterator service_name_it =
+          resource->GetAttributes().find(
+              opentelemetry::sdk::resource::SemanticConventions::kServiceName);
+      opentelemetry::sdk::resource::ResourceAttributes::const_iterator service_namespace_it =
+          resource->GetAttributes().find(
+              opentelemetry::sdk::resource::SemanticConventions::kServiceNamespace);
+
+      if (service_namespace_it != resource->GetAttributes().end() &&
+          service_name_it != resource->GetAttributes().end())
+      {
+        AddPrometheusLabel(kPrometheusJob,
+                           AttributeValueToString(service_namespace_it->second) + "/" +
+                               AttributeValueToString(service_name_it->second),
+                           &metric.label);
+        break;
+      }
+      else if (service_name_it != resource->GetAttributes().end())
+      {
+        AddPrometheusLabel(kPrometheusJob, AttributeValueToString(service_name_it->second),
+                           &metric.label);
+        break;
+      }
+    } while (false);
+
+    opentelemetry::sdk::resource::ResourceAttributes::const_iterator service_instance_id_it =
+        resource->GetAttributes().find(
+            opentelemetry::sdk::resource::SemanticConventions::kServiceInstanceId);
+
+    if (service_instance_id_it != resource->GetAttributes().end())
+    {
+      AddPrometheusLabel(kPrometheusInstance,
+                         AttributeValueToString(service_instance_id_it->second), &metric.label);
+    }
+    else
+    {
+      // Add a empty instance label if it's not exist
+      AddPrometheusLabel(kPrometheusInstance, "", &metric.label);
+    }
+  }
```

Comment:
Please remove this.  SDK exporters should not convert service.* resource attributes to job + instance.  That conversion is done by the prometheus receiver in the otel collector, rather than SDKs.

Reply:
Thanks and done.